File: python/11_Container_With_Most_Water.py
class Solution:
    # @param {integer[]} height
    # @return {integer}
    def maxArea(self, height):
        res,left,right=0,0,len(height)-1
        if right<1:
            return 0
        while left<right:
            res=max(res,(right-left)*min(height[right],height[left]))
            if height[right]>height[left]:
                left=left+1
            elif height[right]<height[left]:
                right=right-1
            else:
                left,right=left+1,right-1
        return res

File: Java/11_Container_With_Most_Water.java
public class Solution {
    public int maxArea(int[] height) {
        if(height.length <= 1)
            return 0;
        int max=0;
        int left=0,right=height.length-1;
        while(left<right){
            max=Math.max(max,(right-left)*Math.min(height[left], height[right]));
            if(height[left]<height[right]){
                left++;
            }else if(height[left]>height[right]){
                right--;
            }else{
                left++;
                right--;
            }
        }
        return max;
    }
    
    public static void main(String[] args){
        Solution sl=new Solution();
        int[][] testdata={
            {
                1,2,3,4,5,4,5,6
            },
            {
                2,3,2,1
            }
        };
        for(int[] s:testdata){
            System.out.println(sl.maxArea(s));
        }
    }
}